File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultMpeg4Gif.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * stas
 * 1/12/16.
 */
public class InlineQueryResultMpeg4Gif extends InlineQueryResult<InlineQueryResultMpeg4Gif> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String mpeg4_url;

    private String thumbnail_url;
    private String thumbnail_mime_type;

    private Integer mpeg4_width;
    private Integer mpeg4_height;
    private Integer mpeg4_duration;

    private String title;
    private String caption;
    private String parse_mode;

    public InlineQueryResultMpeg4Gif(String id, String mpeg4Url, String thumbnailUrl) {
        super("mpeg4_gif", id);
        this.mpeg4_url = mpeg4Url;
        this.thumbnail_url = thumbnailUrl;
    }

    public InlineQueryResultMpeg4Gif mpeg4Width(Integer mpeg4Width) {
        this.mpeg4_width = mpeg4Width;
        return this;
    }

    public InlineQueryResultMpeg4Gif mpeg4Height(Integer mpeg4Height) {
        this.mpeg4_height = mpeg4Height;
        return this;
    }

    public InlineQueryResultMpeg4Gif mpeg4Duration(Integer mpeg4Duration) {
        this.mpeg4_duration = mpeg4Duration;
        return this;
    }

    /**
     * @deprecated Use thumbnailMimeType instead
     */
    @Deprecated 
    public InlineQueryResultMpeg4Gif thumbMimeType(String thumbMimeType) {
        return thumbnailMimeType(thumbMimeType);
    }

    public InlineQueryResultMpeg4Gif thumbnailMimeType(String thumbnailMimeType) {
        this.thumbnail_mime_type = thumbnailMimeType;
        return this;
    }

    public InlineQueryResultMpeg4Gif title(String title) {
        this.title = title;
        return this;
    }

    public InlineQueryResultMpeg4Gif caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultMpeg4Gif parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultMpeg4Gif.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * stas
 * 1/12/16.
 */
class InlineQueryResultMpeg4Gif(id: String, private val mpeg4_url: String, private val thumb_url: String) : InlineQueryResult<InlineQueryResultMpeg4Gif>("mpeg4_gif", id), Serializable {

    private var mpeg4_width: Int? = null
    private var mpeg4_height: Int? = null
    private var mpeg4_duration: Int? = null
    private var title: String? = null
    private var caption: String? = null
    private var parse_mode: String? = null

    fun mpeg4Width(mpeg4Width: Int?): InlineQueryResultMpeg4Gif {
        this.mpeg4_width = mpeg4Width
        return this
    }

    fun mpeg4Height(mpeg4Height: Int?): InlineQueryResultMpeg4Gif {
        this.mpeg4_height = mpeg4Height
        return this
    }

    fun mpeg4Duration(mpeg4Duration: Int?): InlineQueryResultMpeg4Gif {
        this.mpeg4_duration = mpeg4Duration
        return this
    }

    fun title(title: String): InlineQueryResultMpeg4Gif {
        this.title = title
        return this
    }

    fun caption(caption: String): InlineQueryResultMpeg4Gif {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultMpeg4Gif {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
